submissions/mustbefail/memory-pair-game/app.js

```diff
@@ -0,0 +1,117 @@
+const cardNames = [
+  'bear',
+  'bill',
+  'blaster',
+  'diary',
+  'dragon',
+  'fighter',
+];
+
+const state = {
+  flippedCards: {
+    flippedCardsCount: null,
+    flippedCollection: null,
+  },
+  hiddenCardsCount: null,
+  gameState: null
+};
+
+const makeCardpool = () => {
+  const shuffleCards = (cardNames) => {
+    return [...cardNames, ...cardNames]
+      .sort(() => (0.5 - Math.random()));
+  };
+  const shuffledCards = shuffleCards(cardNames);
+  const cardsContainer = document.querySelector('.cards-container');
+
+  const cardTemplate = (cardName) => `
+    <div class="card" data-card-id="${cardName}" data-flip="false">
+    <div class="card-back">
+      <img src="./resources/gf-pics/${cardName}.png" alt="flag" class="card-img" />
+    </div>
+    <div class="card-front">
+    </div>
+    </div>
+  `;
+
+  cardsContainer.innerHTML = shuffledCards.map(cardTemplate).join('');
+
+  state.gameState = 'in progress';
+  
+  const cards = document.querySelectorAll('.card');
+  cards.forEach((el) => el.addEventListener('click', handler));
+};
+
+const updateCardsState = () => {
+  const flippedCards = document.querySelectorAll('[data-flip="true"]');
+  state.flippedCards.flippedCardsCount = flippedCards.length;
+  state.flippedCards.flippedCollection = [...flippedCards];
+};
+
+const checkFlippedCardsCount = () => state.flippedCards.flippedCardsCount === 2;
+
+const checkIdentity = () => {
+  const [first, second] = state.flippedCards.flippedCollection.map(el => el.dataset.cardId);
+  return first === second;
+};
+
+const CheckWinCondition = () => document.querySelectorAll('.hidden').length === 12;
+
+const flipCard = (card) => {
+  card.classList.add('flip');
```

Comment:
I have two problems with this function:  1) it is called `flipCard` and receives one card as an argument, but under the hood is affects all cards (by calling updateCardState) 2) iw does two separate things - flip one card and update all the others, while ideally a function should do only one thing.

I would try to remove `updateCardsState()` call from this function and give this responsibility to some other entity